OSDOCS-6062: Steps to create a NodeFeatureDiscovery instance using the CLI for disconnected environment

abrennan89 opened the pull request:
Version(s):
4.12+

Issue:
https://issues.redhat.com/browse/OSDOCS-6062

Link to docs preview:
https://78787--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hardware_enablement/psap-node-feature-discovery-operator.html#psap-node-feature-discovery-operator-using-the-operator

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
The previous version of these docs contained 2 procedures inside one module file, so I have split these into 2 modules for CLI and web console, and added a third module for CLI in a disconnected env.

chr15p reviewed (approved):
in a couple of places we're pointing to an old version of the upstream docs (https://kubernetes-sigs.github.io/node-feature-discovery/v0.10/advanced/worker-configuration-reference.html where v0.16.2 is the latest version)   should we not link to the OCP docs   [e.g. here](https://docs.openshift.com/container-platform/4.16/hardware_enablement/psap-node-feature-discovery-operator.html#configuring-the-node-feature-discovery_node-feature-discovery-operator)  ?  or the latest upstream?
or is the old docs because its for 4.12 (they are still valid for later versions)?
apart from that minor niggle lgtm

yevgeny-shnaidman commented:
looks good to me to (besides the issue raised by @chr15p ). One question though: why are the files all prefixed with "psap"? PSAP team was responsible for NFD previously, but now NFD is under edge  group. Maybe we can skip prefixes altogether?

abrennan89 commented:
> looks good to me to (besides the issue raised by @chr15p ). One question though: why are the files all prefixed with "psap"? PSAP team was responsible for NFD previously, but now NFD is under edge group. Maybe we can skip prefixes altogether?

This was just to keep consistency with the existing file named `hardware_enablement/psap-node-feature-discovery-operator.adoc`, but I can update the naming. Thanks!

abrennan89 commented:
> in a couple of places we're pointing to an old version of the upstream docs (https://kubernetes-sigs.github.io/node-feature-discovery/v0.10/advanced/worker-configuration-reference.html where v0.16.2 is the latest version) should we not link to the OCP docs [e.g. here](https://docs.openshift.com/container-platform/4.16/hardware_enablement/psap-node-feature-discovery-operator.html#configuring-the-node-feature-discovery_node-feature-discovery-operator) ? or the latest upstream? or is the old docs because its for 4.12 (they are still valid for later versions)? apart from that minor niggle lgtm

On second look, the OCP docs we're referencing for this are immediately after the procedures in this PR, so I have just removed the outdated links. I don't think there's a real need here to link to OCP docs immediately after the procedure on the same page, especially since the OCP doc guidelines would mean needing to do this outside the module files (not in the place where the current upstream docs link is).

abrennan89 commented:
> looks good to me to (besides the issue raised by @chr15p ). One question though: why are the files all prefixed with "psap"? PSAP team was responsible for NFD previously, but now NFD is under edge group. Maybe we can skip prefixes altogether?

For the new files that I have added, I have renamed them to remove the prefix. However, for existing files it would be out of scope to rename them, since this would require fixes for all existing links etc. If you want to have the prefixes removed from all the files that currently have `psap`, please discuss this with your usual docs writers to scope this as a separate Jira clean up task.

lahinson commented on hardware_enablement/psap-node-feature-discovery-operator.adoc:
```diff
@@ -12,7 +12,16 @@ include::modules/psap-node-feature-discovery-operator.adoc[leveloffset=+1]
+[id="psap-node-feature-discovery-operator-using-the-operator"]
+== Using the Node Feature Discovery Operator
+
+The Node Feature Discovery (NFD) Operator orchestrates all resources needed to run the Node-Feature-Discovery daemon set by watching for a `NodeFeatureDiscovery` custom resource (CR). Based on the `NodeFeatureDiscovery` CR, the Operator creates the operand (NFD) components in the desired namespace. You can edit the CR to use another namespace, image, image pull policy, and `nfd-worker-conf` confi ...
```
```suggestion
The Node Feature Discovery (NFD) Operator orchestrates all resources needed to run the Node-Feature-Discovery daemon set by watching for a `NodeFeatureDiscovery` custom resource (CR). Based on the `NodeFeatureDiscovery` CR, the Operator creates the operand (NFD) components in the selected namespace. You can edit the CR to use another namespace, image, image pull policy, and `nfd-worker-conf` config map, among other options.
```
IBM Style word usage warns against using the term "desire" or "desired" in technical documentation. I suggested "selected" here, but feel free to replace that if it's not quite right.

lahinson commented on modules/creating-nfd-cr-cli-disconnected.adoc:
```diff
@@ -0,0 +1,167 @@
+[id="creating-nfd-cr-cli-disconnected_{context}"]
+= Create a NodeFeatureDiscovery CR by using the CLI in a disconnected environment
+
+As a cluster administrator, you can create a `NodeFeatureDiscovery` CR instance using the {oc-first}.
```
```suggestion
As a cluster administrator, you can create a `NodeFeatureDiscovery` CR instance by using the {oc-first}.
```
From the _using_ entry in IBM Style word usage: "To avoid ambiguity, replace this gerund with either _by using_ or _that uses_."

lahinson commented on modules/creating-nfd-cr-cli-disconnected.adoc:
```diff
@@ -0,0 +1,167 @@
+.Prerequisites
+
+* You have access to an {product-title} cluster
+* You have installed the {oc-first}.
```
```suggestion
* You installed the {oc-first}.
```
Revised to avoid the present perfect tense and use the simple past tense, per IBM Style.

lahinson commented on modules/creating-nfd-cr-cli-disconnected.adoc:
```diff
@@ -0,0 +1,167 @@
+
+* You have access to an {product-title} cluster
+* You have installed the {oc-first}.
+* You have logged in as a user with `cluster-admin` privileges.
```
```suggestion
* You logged in as a user with `cluster-admin` privileges.
```

lahinson commented on modules/creating-nfd-cr-cli-disconnected.adoc:
```diff
@@ -0,0 +1,167 @@
+* You have access to an {product-title} cluster
+* You have installed the {oc-first}.
+* You have logged in as a user with `cluster-admin` privileges.
+* You have installed the the NFD Operator.
```
```suggestion
* You installed the the NFD Operator.
```

lahinson commented on modules/creating-nfd-cr-cli-disconnected.adoc:
```diff
@@ -0,0 +1,167 @@
+  name: nfd-instance
+spec:
+  operand:
+    image: <mirror_registry>/openshift4/ose-node-feature-discovery@<image_digest>
```
Does this need a callout? Or will most customers know what to enter here?

abrennan89 replied:
I think it's OK since it's sort of a set value except for the user's mirror registry value and image digest value which are given as variables

lahinson commented on modules/creating-nfd-cr-cli-disconnected.adoc:
```diff
@@ -0,0 +1,167 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="creating-nfd-cr-cli-disconnected_{context}"]
+= Create a NodeFeatureDiscovery CR by using the CLI in a disconnected environment
```
```suggestion
= Creating a NodeFeatureDiscovery CR by using the CLI in a disconnected environment
```
Procedure titles begin with gerunds.

lahinson commented on modules/creating-nfd-cr-cli.adoc:
```diff
@@ -3,29 +3,23 @@
-[id="using-the-node-feature-discovery-operator_{context}"]
-= Using the Node Feature Discovery Operator
+[id="creating-nfd-cr-cli_{context}"]
+= Create a NodeFeatureDiscovery CR by using the CLI
```
```suggestion
= Creating a NodeFeatureDiscovery CR by using the CLI
```

lahinson commented on modules/creating-nfd-cr-cli.adoc:
```diff
@@ -3,29 +3,23 @@
-* Log in as a user with `cluster-admin` privileges.
-* Install the NFD Operator.
+* You have access to an {product-title} cluster
+* You have installed the {oc-first}.
```
```suggestion
* You installed the {oc-first}.
```

lahinson commented on modules/creating-nfd-cr-cli.adoc:
```diff
@@ -3,29 +3,23 @@
-* Install the NFD Operator.
+* You have access to an {product-title} cluster
+* You have installed the {oc-first}.
+* You have logged in as a user with `cluster-admin` privileges.
```
```suggestion
* You logged in as a user with `cluster-admin` privileges.
```

lahinson commented on modules/creating-nfd-cr-cli.adoc:
```diff
@@ -110,13 +104,11 @@ spec:
+$ oc apply -f <filename>
 ----
 
 .Verification
```
On line 116, revise the wording to "To verify that the instance is created, run the following command:"

lahinson commented on modules/creating-nfd-cr-web-console.adoc:
```diff
@@ -0,0 +1,23 @@
+.Procedure
+
+. Navigate to the *Operators* -> *Installed Operators* page.
+. Find *Node Feature Discovery* and see a box under *Provided APIs*.
```
I'm not sure what the UI looks like here, but this seems like an odd step to me. Should this be combined with the next step to say something like this? "In the *Node Feature Discovery* section, under *Provided APIs*, click *Create instance*."

lahinson reviewed (commented):
Looks good! I made a few suggestions for your consideration.

abrennan89 commented:
/cherrypick enterprise-4.12

abrennan89 commented:
/cherrypick enterprise-4.13

abrennan89 commented:
/cherrypick enterprise-4.14

abrennan89 commented:
/cherrypick enterprise-4.15

abrennan89 commented:
/cherrypick enterprise-4.16

abrennan89 commented:
/cherrypick enterprise-4.17
